scenario_player/runner.py

```diff
@@ -153,289 +335,326 @@ def local_seed(self) -> str:
-
-        self.node_controller.initialize_nodes()
-        node_addresses = self.node_controller.addresses
-        node_count = len(self.node_controller)
-        balance_per_node = {
-            address: self.client.balance(to_canonical_address(address))
-            for address in node_addresses
-        }
-        low_balances = {
-            address: balance
-            for address, balance in balance_per_node.items()
-            if balance < NODE_ACCOUNT_BALANCE_MIN
-        }
-        log.debug("Node eth balances", balances=balance_per_node, low_balances=low_balances)
-        if low_balances:
-            log.info("Funding nodes", nodes=low_balances.keys())
-            for address, balance in low_balances.items():
-                params = {
-                    "client_id": self.definition.spaas.rpc.client_id,
-                    "to": address,
-                    "value": NODE_ACCOUNT_BALANCE_FUND - balance,
-                    "startgas": 21_000,
-                }
-                resp = self.service_session.post("spaas://rpc/transactions", json=params)
-                tx_hash = resp.json()["tx_hash"]
-                fund_tx.add(tx_hash)
+        token_definition = self.definition.token
+        reuse_token_from_file = token_definition.reuse_token
+        token_address = to_canonical_address(token_definition.address)
+        deploy_new = not (reuse_token_from_file or token_address)
+        token_data_path = str(self.data_path.joinpath("token.info"))
+
+        if deploy_new:
+            contract = proxy_manager.contract_manager.get_contract(CONTRACT_HUMAN_STANDARD_TOKEN)
+            contract_proxy, receipt = self.client.deploy_single_contract(
+                contract_name=CONTRACT_HUMAN_STANDARD_TOKEN,
+                contract=contract,
+                constructor_parameters=(
+                    ORCHESTRATION_MAXIMUM_BALANCE,
+                    token_definition.name,
+                    token_definition.decimals,
+                    token_definition.symbol,
+                ),
+            )
+            token_address = to_canonical_address(contract_proxy.address)
 
-        try:
-            for node_runner in self.node_controller._node_runners:
-                node_runner.start()
-        except Exception:
-            log.error("failed to start", exc_info=True)
-            raise
-
-        def wait():
-            with gevent.Timeout(MAX_RAIDEN_STARTUP_TIME):
-                for node_runner in self.node_controller._node_runners:
-                    url = f"http://{node_runner.base_url}/api/v1/address"
-                    while True:
-                        try:
-                            if self.session.get(url).ok:
-                                break
-                        except requests.exceptions.RequestException:
-                            pass
-                        gevent.sleep(0.5)
-            log.info("All nodes ready")
+            if reuse_token_from_file:
```

Comment:
This can never be true unless I'm reading the code wrong.

Reply:
humm, it should be the same as this:

https://github.com/raiden-network/scenario-player/blob/edd041e816776c141a1182f089ab6bb75a976e91/scenario_player/utils/token.py#L265-L269

no?

Reply:
I guess I can change the code to something like:
```python
if token_address:
  # instantiate proxy
elif reuse_token and token_info.exists():
  # load json
else:
  token = deploy()
  if reuse: save_json()
```